### PR TITLE
Issue 2391: Apply security checks for `listScopes` API

### LINKED ISSUE #2391: Pravega security: `listScopes` should do authorization checks too
**Problem description**
Current implementation of `listScopes` does authorize or authenticate. This should follow the usual
procedure where we authorize the user for each stream and just show the stream that he has access to.

**Problem location**
`listScopes` implementation here:
https://github.com/pravega/pravega/blob/master/controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java#L469

**Suggestions for an improvement**
This should follow the usual
procedure where we authorize the user for each stream and just show the stream that he has access to.

### PR DESCRIPTION
**Pull request by arvindkandhare**
**Change log description**
implement validation for `listScopes` API.
This API is called from the REST interface.  The existing behavior is to allow this call to go through for all the callers. 
This change ensures that the user access is validated for the API. Once the user has access to the root, only the scopes for which the user had access are returned. For this API to return a scope following constraints should be true:

1. It should be a valid user
2. The user should have at least read access to the given scope.
 
**Purpose of the change**
This fixes #2391.

**What the code does**
Ensures that the user is validated and only the valid scopes are returned.
 
**How to verify it**
`testListScopes` test case passes in all the three scenarios: `StreamMetadataTests` - without auth, `SecureStreamMetadataTests` - auth enabled with valid auth and `FailingSecureStreamMetadataTests` - auth enabled with no access pass.

### CODE REVIEW

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java [hunk 1]
```diff
@@ -71,23 +74,25 @@ public String getHandlerName() {
...
-    public boolean authenticate(String token) {
+    public Principal authenticate(String token) throws AuthException {
         String[] parts = parseToken(token);
         String userName = parts[0];
         String password = parts[1];
 
         try {
-            return userMap.containsKey(userName) && encryptor.checkPassword(password, userMap.get(userName).encryptedPassword);
+            if (userMap.containsKey(userName) && encryptor.checkPassword(password, userMap.get(userName).encryptedPassword)) {
+                return new UnixPrincipal(userName);
```
Comment (EronWright) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: Can't use `UnixPrincipal` because `com.sun.*` has issues in JDK9.  

I suggest simply introducing a `DefaultPrincipal` that simply carries the username.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java [hunk 2]
```diff
@@ -71,23 +74,25 @@ public String getHandlerName() {
...
+            }
         } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
             log.warn("Excpetion during password authentication", e);
-            return false;
+            throw new AuthenticationException(e);
         }
+        return null;
```
Comment (EronWright) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: What are the semantics of `null` here?   A good practice would be to avoid unnecessary information disclosure, e.g. whether a given user exists.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 3]
```diff
@@ -462,16 +462,32 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
...
-                }).exceptionally(exception -> {
-                        log.warn("listScopes failed with exception: " + exception);
-                        return Response.status(Status.INTERNAL_SERVER_ERROR).build();
-                }).thenApply(asyncResponse::resume)
-                .thenAccept(x ->  LoggerHelpers.traceLeave(log, "listScopes", traceId));
+                         .thenApply(scopesList -> {
+                             ScopesList scopes = new ScopesList();
+                             scopesList.forEach(scope -> {
+                                 try {
+                                     authenticate(scope, READ);
```
Comment (EronWright) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Could you please decouple `authenticate` and `authorize` so that `authenticate` is called once?   I thought that was the main point of the interface change.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java (hunk 1 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: Can you point a reference to the issues? `UnixPrincipal` seems too entrenched to be deprecated!

Comment (EronWright) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: This is related to JDK9 modularization ("jigsaw"). You're right that I'm being overly conservative in the case of `UnixPrincipal`; the `jdk.security.auth` module seems to export it for normal use.  
https://docs.oracle.com/javase/9/docs/api/jdk.security.auth-summary.html

Still feels wrong to me, but not a big deal.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 3 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed. My interpretation of the discussions was that we were planning to optimize this later. Updated with decoupled implementation.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java (hunk 2 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: Fixed. Throwing an exception when the user does not exist.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 4]
```diff
@@ -141,7 +142,40 @@ private void authenticate(String resourceName, AuthHandler.Permissions level) th
             String credentials = authParams.get(0);
             assert credentials != null;
 
-            if (!pravegaAuthManager.authenticate(resourceName, credentials, level)) {
+            if (!pravegaAuthManager.authenticateAuthorize(resourceName, credentials, level)) {
+                throw new AuthorizationException("Auth failed for " + resourceName, 403);
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Isn't there a class somewhere that defines these response codes somewhere? I don't like having hardcoded numbers throughout the codebase.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 5]
```diff
@@ -141,7 +142,40 @@ private void authenticate(String resourceName, AuthHandler.Permissions level) th
...
+    private void authorize(String resourceName, Principal principal, AuthHandler.Permissions level) throws AuthException {
+        if (pravegaAuthManager != null ) {
+            List<String> authParams = headers.getRequestHeader(HttpHeaders.AUTHORIZATION);
+
+            if (authParams == null || authParams.isEmpty()) {
+                throw new AuthenticationException("Auth failed for " + resourceName);
+            }
+
+            String credentials = authParams.get(0);
+            assert credentials != null;
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Why is this an assert? Isn't this a parameter, in which case it should be a precondition check?

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 6]
```diff
@@ -141,7 +142,40 @@ private void authenticate(String resourceName, AuthHandler.Permissions level) th
...
+
+            if (!pravegaAuthManager.authorize(resourceName, principal, credentials, level)) {
                 throw new AuthorizationException("Auth failed for " + resourceName, 403);
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: hardcoded number

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 7]
```diff
@@ -462,16 +496,34 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
     public void listScopes(final SecurityContext securityContext, final AsyncResponse asyncResponse) {
         long traceId = LoggerHelpers.traceEnter(log, "listScopes");
 
+        Principal principal;
+        try {
+            principal = authenticate();
+            authorize("/", principal, READ);
+        } catch (AuthException e) {
+            log.warn("Get scopes failed due to authentication failure.");
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: add `e` as an argument

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 8]
```diff
@@ -462,16 +496,34 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
...
-                }).thenApply(asyncResponse::resume)
-                .thenAccept(x ->  LoggerHelpers.traceLeave(log, "listScopes", traceId));
+                         .thenApply(scopesList -> {
+                             ScopesList scopes = new ScopesList();
+                             scopesList.forEach(scope -> {
+                                 try {
+                                     authorize(scope, principal, READ);
+                                     scopes.addScopesItem(new ScopeProperty().scopeName(scope));
+                                 } catch (AuthException e) {
+                                     log.warn("Not adding {} to list because authentication exception {}", scope, e);
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: No need for second `{}` - that will just double-log the exception

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 9]
```diff
@@ -462,16 +496,34 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
...
+                                 }
+                             });
+                             return Response.status(Status.OK).entity(scopes).build();
+                         }).exceptionally(exception -> {
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Please rework this method's indentation. The next line starts before the upper line which makes this very hard to read

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 10]
```diff
@@ -462,16 +496,34 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
...
+            log.warn("listScopes failed with exception: " + exception);
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: pass `exception` as an argument; don't concat it

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java [hunk 11]
```diff
@@ -462,16 +496,34 @@ public void listReaderGroups(final String scopeName, final SecurityContext secur
...
+            return Response.status(Status.INTERNAL_SERVER_ERROR).build();
+        }).thenApply(asyncResponse::resume)
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: You can merge these two callbacks into one.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 12]
```diff
@@ -69,16 +71,66 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
...
+        Preconditions.checkNotNull(credentials, "credentials");
+        boolean retVal = false;
+        String[] parts = credentials.split("\\s+", 2);
+        if (parts.length != 2) {
+            throw new AuthenticationException("Malformed request");
+        }
+        String method = parts[0];
+        String token = parts[1];
+        AuthHandler handler = getHandler(method);
+        assert handler != null;
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: IMO, this should be a precondition check, not assert.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 13]
```diff
@@ -69,16 +71,66 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
...
+            throw new AuthenticationException("Authentication failure");
         }
         return retVal;
     }
 
+    /**
+     *
+     * API to authenticate a given credential.
+     * @param credentials  Credentials used for authentication.
+     * @return         Returns the Principal if the entity represented by credentials is authenticated.
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: indentation

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 14]
```diff
@@ -69,16 +71,66 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
...
+        AuthHandler handler = getHandler(method);
+        assert handler != null;
+        return handler.authenticate(token);
+    }
+
+    /**
+     *
+     * API to authorize a given principal and credential.
+     *
+     * @param resource The resource identifier for which the access needs to be controlled.
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Please align the `@param` texts; the IDE should do it for you

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 15]
```diff
@@ -69,16 +71,66 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
...
+    public boolean authorize(String resource, Principal principal, String credentials, AuthHandler.Permissions level) throws AuthException {
+        Preconditions.checkNotNull(credentials, "credentials");
+        boolean retVal = false;
+        String[] parts = credentials.split("\\s+", 2);
+        if (parts.length != 2) {
+            throw new AuthenticationException("Malformed request");
+        }
+        String method = parts[0];
+        AuthHandler handler = getHandler(method);
+        assert handler != null;
```
Comment (andreipaduroiu) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: precondition check

Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java (hunk 14 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed. IDE did not do it for me :). Did by hand.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java (hunk 13 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java (hunk 12 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed. Added a `Preconditions` check.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 11 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed. Merged into a single callback.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 8 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 7 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 6 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 5 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: This is not a parameter. Replaced assert with a `Preconditions.notNull` check.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 4 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed. Using HTTP Status.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 10 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 9 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 16]
```diff
@@ -69,16 +71,66 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
...
+     * @param credentials  Credentials used for authentication.
+     * @param level    Expected level of access.
+     * @param principal Principal associated with the credentials.
+     *
+     * @return Returns true if the entity represented by the credentials has given level of access to the resource.
+     *      Returns false if the entity does not have access.
+     * @throws AuthException if an authentication failure occurred.
+     */
+    public boolean authorize(String resource, Principal principal, String credentials, AuthHandler.Permissions level) throws AuthException {
+        Preconditions.checkNotNull(credentials, "credentials");
```
Comment (sarlaccpit) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Maybe we could have a field in the principal object that indicates which method was used.   This way:
- we don't have to pass credentials again here for the sole purpose of extracting again the method
- it gives the Principal a tighter connection between the authenticate and authorize calls.  Meaning the method has been extracted once, kept in one place and reused.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 5 above)
Comment (andreipaduroiu) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: I believe this is still there.

Why would it not be considered a parameter if you fetch it from `authParams` which is retrieved externally?

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 17]
```diff
@@ -69,16 +71,67 @@ public boolean authenticate(String resource, String credentials, AuthHandler.Per
...
+    /**
+     *
+     * API to authenticate a given credential.
+     * @param credentials  Credentials used for authentication.
+     *
+     * @return Returns the Principal if the entity represented by credentials is authenticated.
+     * @throws AuthException if an authentication failure occurred.
+     */
+    public Principal authenticate(String credentials) throws AuthException {
+        Preconditions.checkNotNull(credentials, "credentials");
```
Comment (sarlaccpit) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: line 96-104 can be moved to a common convenience function used by authenticate and authorize

Comment (sarlaccpit) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: oh right.  I guess lines 96-101 then.   The function would just return a String[].  (this is minor).

Also retVal doesn't seem to be used.

Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed. Refactored the code and dropped `retVal`.

File: controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java (hunk 5 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java]: Fixed. Using `PreConditions`.

File: client/src/main/java/io/pravega/client/netty/impl/RawClient.java [hunk 18]
```diff
@@ -9,10 +9,10 @@
  */
 package io.pravega.client.netty.impl;
 
+import io.pravega.auth.AuthenticationException;
```
Comment (fpj) [inline client/src/main/java/io/pravega/client/netty/impl/RawClient.java]: There are a bunch of gratuitous changes to the order of imports in this PR. Please avoid such unnecessary formatting changes, which are unrelated to the issue being resolved here.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java (hunk 1 above)
Comment (fpj) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: Please create an issue for this if you don't want to resolve here.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 19]
```diff
@@ -57,28 +59,77 @@ private AuthHandler getHandler(String handlerName) throws AuthenticationExceptio
      *                 Returns false if the entity does not have access. 
      * @throws AuthenticationException if an authentication failure occurred.
      */
-    public boolean authenticate(String resource, String credentials, AuthHandler.Permissions level) throws AuthenticationException {
+    public boolean authenticateAuthorize(String resource, String credentials, AuthHandler.Permissions level) throws AuthenticationException {
```
Comment (fpj) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Style comment: it should be `authenticateAndAuthorize`, no?

File: shared/authplugin/src/main/java/io/pravega/auth/AuthConstants.java [hunk 20]
```diff
@@ -7,7 +7,7 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  */
-package io.pravega.common.auth;
+package io.pravega.auth;
```
Comment (fpj) [inline shared/authplugin/src/main/java/io/pravega/auth/AuthConstants.java]: This is another change unrelated to this PR. Let's not interleave formatting changes in PRs that are not refactoring the given code. In fact, I want to understand why we are moving this class and others in this PR from `common` to `shared`. Is there even an issue for that and has it been discussed?

Comment (arvindkandhare) [reply shared/authplugin/src/main/java/io/pravega/auth/AuthConstants.java]: The interface `AuthHandler` had changed as a response to a comment on the PR. Which mandated changing of the location of the exceptions in this package. As with the current changes, they needed to be visible to the plugin implementation. The interface change was discussed and the move of the exceptions to new packages is implicit as part of the interface change.
Once the exceptions are moved, then there is only class left in this package and I moved that too. Do let me know whether this justifies the move for this particular class.

File: client/src/main/java/io/pravega/client/netty/impl/RawClient.java (hunk 18 above)
Comment (arvindkandhare) [reply client/src/main/java/io/pravega/client/netty/impl/RawClient.java]: This exception is moved to a different package because of change in the `AuthHandler` interface. These were done as part of a review comment. With Pravega settings the IDE sorts the imports alphabetically. Because of this the location of this import changed. Do let me know if you think this change should still be inline.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java (hunk 19 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed.

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java (hunk 1 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PasswordAuthHandler.java]: Here is an issue to track this: https://github.com/pravega/pravega/issues/2821

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java [hunk 21]
```diff
@@ -108,4 +159,6 @@ public void registerInterceptors(ServerBuilder<?> builder) {
             log.warn("Exception while loading the auth handlers", e);
         }
     }
+
```
Comment (shiveshr) [inline controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: nit: unnecessary lines

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java (hunk 17 above)
Comment (shiveshr) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: nit: retVal is still there

File: controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java (hunk 21 above)
Comment (arvindkandhare) [reply controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthManager.java]: Fixed.
